crypto/evp/evp_lib.c

```diff
@@ -316,6 +317,10 @@ int EVP_MD_size(const EVP_MD *md)
         EVPerr(EVP_F_EVP_MD_SIZE, EVP_R_MESSAGE_DIGEST_IS_NULL);
         return -1;
     }
+
+    if (md->prov != NULL && md->size != NULL)
+        return (int)md->size();
+
```

Comment:
Does this need a legacy to be removed comment?
The _if_ above looks like it will change too.

Reply:
No - this is non-legacy code!